client/src/store/slices/mentor/index.ts

```diff
@@ -77,7 +78,12 @@ export const saveMentor = createAsyncThunk(
         );
         return editedData;
       } catch (err) {
-        return err.response.data;
+        if (axios.isAxiosError(err)) {
+          console.error(err.response?.data);
+          return rejectWithValue(err.response?.data);
+        } else {
+          console.log("Unexpected error", err);
```

Comment:
you need to also rejectWithValue in the else statement, I know you're already handling this in your other branch so might want to wait for the other branch to merge with main first and then merge, or copy your changes to this branch